### PR TITLE
Spring boot category

### PR DESCRIPTION
**Pull request by dizizzz**

### CODE REVIEW

File: src/main/java/mate/academy/springboot/dto/category/CategoryDto.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+import jakarta.validation.constraints.NotBlank;
+import lombok.Data;
+
+@Data
+public class CategoryDto {
+    private Long id;
+    @NotBlank
+    private String name;
+    private String description;
```
Comment (Sanderfuck) [inline src/main/java/mate/academy/springboot/dto/category/CategoryDto.java:12]: Could you validate description?

**Review by Sanderfuck — CHANGES_REQUESTED** [review]

File: src/main/resources/db/changelog/changes/12-insert-categories.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+                  value: Description 1
+        - insert:
+            tableName: categories
+            columns:
+              - column:
+                  name: name
+                  value: Category 2
+              - column:
+                  name: description
+                  value: Description 2
```
Comment (okuzan) [inline src/main/resources/db/changelog/changes/12-insert-categories.yaml:23]: always make sure file ends with one empty line
```suggestion
                  value: Description 2

```

File: src/main/java/mate/academy/springboot/service/impl/BookServiceImpl.java [hunk 3]
```diff
@@ -70,4 +80,9 @@ public List<BookDto> search(BookSearchParameters params) {
                 .toList();
     }
 
+    @Override
+    public List<Book> findAllByCategoriesId(Long id) {
+        return bookRepository.findAllByCategoriesId(id);
+    }
+
```
Comment (okuzan) [inline src/main/java/mate/academy/springboot/service/impl/BookServiceImpl.java:87]:
```suggestion
```

File: src/main/java/mate/academy/springboot/model/Category.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import lombok.Data;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
+
+@Data
+@Entity
+@SQLDelete(sql = "UPDATE categories SET is_deleted = true WHERE id=?")
+@Where(clause = "is_deleted=false")
```
Comment (okuzan) [inline src/main/java/mate/academy/springboot/model/Category.java:16]:
```suggestion
@SQLDelete(sql = "UPDATE categories SET is_deleted = TRUE WHERE id = ?")
@Where(clause = "is_deleted = FALSE")
```

**Review by okuzan — APPROVED** [review]

**Pull request merged**